cabal-install/src/Distribution/Client/CmdUpdate.hs

```diff
@@ -209,7 +211,7 @@ updateRepo verbosity _updateFlags repoCtxt (repo, indexState) = do
       case updated of
         Sec.NoUpdates  -> do
           now <- getCurrentTime
-          setModificationTime (indexBaseName repo <.> "tar") now
+          setModificationTime (indexBaseName repo <.> "tar") now `catch` (\(e :: SomeException) -> warn verbosity $ "Could not set modification time of index tarball -- " ++ show e)
```

Comment:
This also catches asynchronous exceptions, such as e.g. `UserInterrupt` issued when Ctrl-C is hit; would be more robust to use safe-exception's catch or catch some more targeted set of exceptions, e.g. using `catchIO` as below.